### PR TITLE
Bug 1805840 - Fix layout for add to home screen  with ETP warning.

### PR DESCRIPTION
**Pull request by mcarare**
This adds margins and constraints according to the initial [design](https://www.figma.com/file/2V6wgf9fRFdYfLaf5c3oPj/Product-UI?node-id=2133%3A282824&t=TFcwwSOB4oYYtlBa-0) .

Also added margin to dialog buttons to avoid text overlapping (Figma design does not have margins, the buttons even overlap).

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

**BEFORE:**
English / German (long text) / RTL

  

**AFTER:**
English / German (long text) / RTL

   

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

**Comment by mcarare** [comment]
Test for an even longer text:

**Comment by t-p-white** [comment]
Looks good 👍 
Wonder if there should also be some spacing at the top and bottom?

Screenshot 2022-12-22 at 17 13 17

**Comment by mcarare** [comment]
@t-p-white I also thought of adding that, but I did not want to increase the height (with margins or padding) when presumably all strings would be shorter than the German one. I just added the excessively long text example to illustrate that the layout now expands as needed. (above the previously fixed height of 80 dp)

File: focus-android/app/src/main/res/layout/dialog_add_to_homescreen2.xml [hunk 1]
```diff
@@ -82,6 +87,9 @@
             android:layout_width="24dp"
             android:layout_height="24dp"
             android:layout_gravity="center_vertical"
+            android:layout_marginStart="24dp"
+            android:layout_marginTop="26dp"
+            android:layout_marginEnd="32dp"
```
Comment (t-p-white) [inline focus-android/app/src/main/res/layout/dialog_add_to_homescreen2.xml:92]: I wonder if this value should match marginStart for visual consistency? Appreciate the text is unlikely to be right up against this textview but if it does, should it match the left margin?

**Review by t-p-white — APPROVED** [review]

File: focus-android/app/src/main/res/layout/dialog_add_to_homescreen2.xml (hunk 1 above)
Comment (mcarare) [reply focus-android/app/src/main/res/layout/dialog_add_to_homescreen2.xml]: I've rechecked the designs and that was supposed to be the bottom margin. The margin to ensure space between the icon and text is set by the text view. And it is supposed to be smaller (like in the design) to also align with the title text.

**Comment by mcarare** [comment]
https://github.com/Mergifyio backport releases_v109

**Pull request merged**